src/NHapi.Base/Parser/MessageIterator.cs

```diff
@@ -233,26 +250,23 @@ private void AddNonStandardSegmentAtCurrentPosition()
             Log.Debug(
                 $"Creating non standard segment {direction} on group: {GetCurrentPosition().StructureDefinition.Parent.Name}");
 
-            // TODO: make configurable like hapi?
-            // switch (message.getParser().getParserConfiguration().getUnexpectedSegmentBehaviour())
-            // {
-            //     case ADD_INLINE:
-            //     default:
-            //         parentDefinitionPath = new ArrayList<>(myCurrentDefinitionPath.subList(0, myCurrentDefinitionPath.size() - 1));
-            //         parentStructure = (IGroup)navigateToStructure(parentDefinitionPath);
-            //         break;
-            //     case DROP_TO_ROOT:
-            //         parentDefinitionPath = new ArrayList<>(myCurrentDefinitionPath.subList(0, 1));
-            //         parentStructure = myMessage;
-            //         myCurrentDefinitionPath = myCurrentDefinitionPath.subList(0, 2);
-            //         break;
-            //     case THROW_HL7_EXCEPTION:
-            //         throw new Error(new HL7Exception("Found unknown segment: " + myDirection));
-            // }
-
-            // default hapi behaviour
-            var parentDefinitionPath = new List<Position>(currentDefinitionPath.GetRange(0, currentDefinitionPath.Count - 1));
-            var parentStructure = (IGroup)NavigateToStructure(parentDefinitionPath);
+            List<Position> parentDefinitionPath;
+            IGroup parentStructure;
+
+            switch (parserOptions.UnexpectedSegmentBehaviour)
+            {
+                case UnexpectedSegmentBehaviour.DropToRoot:
+                    parentDefinitionPath = new List<Position>(currentDefinitionPath.GetRange(0, 1));
+                    parentStructure = this.message;
+                    currentDefinitionPath = currentDefinitionPath.GetRange(0, 2);
+                    break;
+                case UnexpectedSegmentBehaviour.ThrowHl7Exception:
+                    throw new HL7Exception($"Found unknown segment: {direction}");
+                default:
```

Comment:
commenting without context here, but...

is this truely the default behaviour? ie: if I added another value to `UnexpectedSegmentBehaviour` would I want this behaviour? If not, would suggest being expicit and including `case: UnexpectedSegmentBehaviour.AddInline` and then throwning an exception as the default case (`UnhandledUnexpectedSegmentBehaviourException` or something)

Reply:
@AMCN41R also that was the default behaviour before the config option as you can see in the diff. all I did here was make it implicitly default (ported from hapi).

Reply:
@AMCN41R I see what you are saying having thought about it, I will make the change.

Reply:
@AMCN41R I had made this change.